usagov/usagov-2021
# USAGOV-1107 FAQ schema

## Jira Task

https://cm-jira.usa.gov/browse/USAGOV-1107

## Description

 - Config files: add an "FAQ Page" checkbox field to Basic Pages.
 - usagov.theme: If the FAQ checkbox is checked, then parse the content to generate question and answer data
 - html.html.twig: If FAQ data is provided, render it in JSONLD FAQ format

## Type of Changes

- [x] Frontend (Twig)
  - Twig creates JSON data
- [x] PHP (usagov.theme)

## Testing Instructions


 - Choose a Basic Page (there is a list of example pages in the Jira ticket)
 - Edit the page to select the "FAQ Page" checkbox
 - Inspect the page and find "FAQPage" JSON data
 - Confirm that the Questions and Answers match the content following the rules described in the Jira ticket

### Requires New Config

### Requires New Content

### Validation Steps
- Test instruction 1
- Test instruction 2
- Test instruction 3

## Security Review




## Reviewer Reminders
- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions
Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
2. Find the commit of this pull request.
3. Build and deploy the changes.
4. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.
